### PR TITLE
add MetaHttpReplaceHandler 

### PR DESCRIPTION
**Pull request by liuyu85cn**
#### brief introduction 
If a user wants to move data from Host A to Host B,

He needs to: 1. move data actually 2. update meta data in meta server 

This feature is to implement step 2 by handle a http request in meta server 

which like 

http://192.168.8.5:12000/replace?from=192.168.8.5&to=66.66.66.66

#### some more details
MetaHttpReplaceHandler will scan all partition information in meta, 

for each partition, it will  replace all the host information if it match argument [from], 

and change to argument [to]

#### corner case warning
Please note this is only a string replacement,

will check if [from] and [to] is in valid url form. 

but there is not guarantee that [to] is an available host

### Why are the changes needed?

dangleptr asked 

### Does this PR introduce any user-facing change?

no

#### error check
check if url doesn't have [from]
check if url doesn't have [to]
check if url has invalid form of [from]
check if url has invalid form of [to]

#### happy path
check if meta data changed according to the request

### CODE REVIEW

**Comment by Amber1990Zhang** [comment]
I guess this is a new way to migrate data. Will it make sense if I put the doc here?
[data migration](https://github.com/vesoft-inc/nebula/blob/master/docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/storage-balance.md)
Please write me a doc draft on [Yuyue](https://www.yuque.com/nebulagraph/emchnb/chohgc).
Thanks for taking care of this.

**Comment by dangleptr** [comment]
@Amber1990Zhang we don't want to make it open for our users.

**Comment by dangleptr** [comment]
Totally,  LGTM.   Well done.

File: src/meta/MetaHttpReplaceHostHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    bool updateSucceed{false};
+    folly::Baton<true, std::atomic> baton;
+    kvstore_->asyncMultiPut(kDefaultSpaceId, kDefaultPartId, std::move(data),
+                        [&] (kvstore::ResultCode code) {
+                            updateSucceed = (code == kvstore::ResultCode::SUCCEEDED);
+                            baton.post();
+                        });
+    baton.wait();
+    return true;
```
[the remark this answers is not in the document]
Comment (liuyu85cn) [reply src/meta/MetaHttpReplaceHostHandler.cpp]: right, fixed and add some err msg, thx

File: src/meta/test/MetaHttpReplaceHandlerTest.cpp [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+DECLARE_int32(ws_storage_http_port);
+
+namespace nebula {
+namespace meta {
+
+
+meta::MetaHttpReplaceHostHandler* gHandler = nullptr;
+kvstore::KVStore*   gKVStore = nullptr;
+std::vector<std::string> dumpKVStore(kvstore::KVStore* p);
```
[the remark this answers is not in the document]
Comment (liuyu85cn) [reply src/meta/test/MetaHttpReplaceHandlerTest.cpp]: right, more specific, fixed thx.

**Comment by dangleptr** [comment]
Please fix the conflicts

**Pull request merged**